modules/@apostrophecms/migration/lib/addMissingSchemaFields.js

```diff
@@ -0,0 +1,141 @@
+const _ = require('lodash');
+const { klona } = require('klona');
+
+module.exports = (self) => {
+  return {
+    async addMissingSchemaFields() {
+      let scans = 0; let updates = 0;
+      const lastPropLists = await self.getLastPropLists();
```

Comment:
A "prop list" is a list of all schema properties that existed, for comparison with the new list for this startup to determine if any property additions might potentially be needed. We make this determination quickly and move on if new work is not needed.